src/common/users.ts

```diff
@@ -499,27 +499,29 @@ export const checkUserStreak = (
 };
 
 export const getLastStreakRecoverDate = async (
-  con: DataSource | EntityManager,
+  con: DataSource,
   userId: string,
 ) => {
-  const lastRecoverAction = await con
-    .getRepository(UserStreakAction)
-    .createQueryBuilder()
-    .select(
-      `MAX(date_trunc('day', usa."createdAt"::timestamptz at time zone COALESCE(u.timezone, 'utc'))::date - interval '1 day')`,
-      'createdAt',
-    )
-    .from(UserStreakAction, 'usa')
-    .innerJoin(DbUser, 'u', 'u.id = usa."userId"')
-    .where(`usa."userId" = :userId`, { userId })
-    .andWhere(`usa.type = :type`, { type: UserStreakActionType.Recover })
-    .getRawOne<UserStreakAction>();
+  const lastRecoverAction = await queryReadReplica(con, ({ queryRunner }) =>
```

Comment:
Since this helper is now used in restore logic, it could potentially make problems due to replica being out of sync. This is called directly in CDC so actually it could be outdated easily.

Reply:
But unlikely it's calculating your streak while you just read is what I meant?
Happy to move it back to main worker though.

Reply:
I am worried for `onUserStreakChange` cdc callback, there on streak change we read from replica which could be outdated due to change just happened on `master`. I agree read replica can be used on gql side but we can maybe fix it by using different helper in CDC.

Maybe in `userStreak` query we can wrap `checkAndClearUserStreak` call in `queryReadReplica` instead of using it inside the helper.

Reply:
Something like this:

```ts
const hasClearedStreak = await queryReadReplica(
  ctx.con,
  ({ queryRunner }) => {
    return checkAndClearUserStreak(queryRunner.connection, info, streak);
  },
);
```